server/market/bookrouter.go

```diff
@@ -559,13 +557,16 @@ func (r *BookRouter) sendBook(conn comms.Link, book *msgBook, msgID uint64) {
 
 // msgOrderBook returns current (latest) order book snapshot.
 func (r *BookRouter) msgOrderBook(book *msgBook) *msgjson.OrderBook {
-	book.mtx.RLock()
-	defer book.mtx.RUnlock()
-
-	if !book.running {
+	// Don't want to block client requests for too long, so return fast if order book
+	// isn't initialized yet instead of proceeding to wait on mutex below for who
+	// knows how long.
+	if !book.running.Load() {
 		return nil
 	}
```

Comment:
If this was the original intent (see the comment I added there) for having `running` flag there, it didn't work because client request would block on `book.mtx` **before** order book is available or not. Or is there any other purpose for `running` ?

Moved `running` from under `book.mtx.RLock()/book.mtx.RUnlock()` section for now.